src/refnux.coffee

```diff
@@ -50,6 +50,15 @@ createStore = (state = {}) ->
         # execute the action
         try
             newval = action state, dispatch
+
+            if newval and typeof(newval.then) is 'function'
+                dispatching = false
+                return newval.then(
```

Comment:
I added this but i am not entirely sure wether we should return the action value or the new state somehow.